src/main/kotlin/no/nav/helse/sykmelding/SykmeldingConsumer.kt

```diff
@@ -27,13 +28,16 @@ class SykmeldingConsumer(streamsBuilder: StreamsBuilder, private val probe: Sykm
     fun build(builder: StreamsBuilder): StreamsBuilder {
         builder.stream<String, JsonNode>(topics, Consumed.with(Serdes.String(), JsonNodeSerde(objectMapper))
                 .withOffsetResetPolicy(Topology.AutoOffsetReset.EARLIEST))
+                .mapValues { jsonNode ->
+                    Sykmelding(jsonNode)
+                }
                 .foreach(::håndterSykmelding)
 
         return builder
     }
 
-    private fun håndterSykmelding(key: String, sykmelding: JsonNode) {
-        probe.mottattSykmelding(key, sykmelding)
+    private fun håndterSykmelding(key: String, sykmelding: Sykmelding) {
```

Comment:
trenger man enda et lag med indirection her. Kunne man kalt proben fra foreach-en

Reply:
jeg tenkte at consumer-klassen skulle ha logikken for å håndtere meldingene, og at byggingen av stream-topologien ble så enkel som mulig. Akkurat det med å logge "mottatte meldinger" burde kanskje gjøres direkte i topologien, siden det burde skje såpass tidlig uansett (og avkoblet selve behandlingen av meldingen).

Kanskje en `peek { probe.mottattSykmelding(key, sykmelding) }` før foreachen?

Reply:
jeg bare synes at en håndter*() metode som bare gjør en ting virker unødvendig

Reply:
enig. dette var mest for å få opp et skall, siden businesslogikken fortsatt er litt uklar. Fokuset har vært å definere yter-rammene, dvs. teste outside-in først.